webapp/resources/templates/fragments/loginProviders.html

```diff
@@ -22,7 +22,7 @@ <h3 th:utext="#{screen.welcome.label.loginwith}"/>
                     <span th:case="linkedin" class="fa fa-linkedin"></span>
                     <span th:case="facebook" class="fa fa-facebook-square"></span>
                     <span th:case="oidc" class="fa fa-openid"></span>
-                    <span th:case="saml2" class="fa fa-lock"></span>
+                    <span th:case="saml2" th:class="${entry.cssClass}" ></span>
```

Comment:
I'm not sure I understand the context here… `#strings.replace(...)` is a sort of *replace all occurrences* function, but it does not allow regex, if that's what you are asking for. Anyway it is not a wrapper around Java's `String#replace` or `String#replaceAll`, but a custom implementation.

The reason for this is that in most cases *replace all occurrences* (as a functionality) is used without any need for regex. But `String#replace` and especially `String#replaceAll` have huge hidden costs in that they create **and compile** a regex `java.util.Pattern` each time they are called. If this happens a significative number of times, it can be a *huge* performance killer in any app (well, not really in the case of `replace` because such Pattern is in `LITERAL` mode, but still not 100% ideal).

So in order to avoid nasty surprises at the view side we left `#strings.replace(...)` implemented as a *replace all occurrences* function with no regex capabilities. Right now there is no regex-capable equivalent, and to be honest I would have some reserves about adding such a function if it means compiling a `Pattern` with each execution, as this type of cost is something unknown for many users (and would be a nasty surprise for the undocumented user).

Anyway, I might not be completely understanding the thing here… is regex what you need? note that if it is a matter of adding CSS classes to existing values you could use `th:classappend` or `th:classprepend`...